pkg/watch/manager.go

```diff
@@ -223,26 +223,32 @@ func (wm *Manager) doRemoveWatch(r *Registrar, gvk schema.GroupVersionKind) erro
 	return nil
 }
 
-// replaceWatches ensures all and only desired watches are running.
-func (wm *Manager) replaceWatches(r *Registrar) error {
+// reconcileWatches ensures all and only desired watches are running.
+func (wm *Manager) reconcileWatches() error {
 	wm.watchedMux.Lock()
 	defer wm.watchedMux.Unlock()
 
 	var errlist errorList
 
 	desired := wm.managedKinds.Get()
-	for gvk := range wm.watchedKinds {
-		if _, ok := desired[gvk]; !ok {
+	for gvk, v := range wm.watchedKinds {
+		if _, ok := desired[gvk]; ok {
+			// Still in the desired set.
+			continue
+		}
+		for r := range v.registrars {
 			if err := wm.doRemoveWatch(r, gvk); err != nil {
-				errlist = append(errlist, fmt.Errorf("removing watch for %+v %w", gvk, err))
+				errlist = append(errlist, fmt.Errorf("removing watch for %+v (%s) %w", gvk, r.parentName, err))
 			}
 		}
 	}
 
 	// Add desired watches. This is idempotent for existing watches.
-	for gvk := range desired {
```

Comment:
I'm not clear what bug this was addressing?

Reply:
NVM, I see the bug is described in the top of the PR. In this case it's a different side of the same issue. We are attempting to add watches where they don't belong, though this seems like more of an annoyance than anything, as it appears we check for this eventuality and avoid taking action in `doAddWatch`.

Trying to modify all watches wholesale may be tricky as we are relying on responses from calls to Add()/Remove() elsewhere in the project to trigger signaling. I don't think it would break things currently, but the mixing of paradigms is a warning signal.

Could we instead just filter out those entries in desired that don't have this registrar listed as a member?

Reply:
> we are relying on responses from calls to Add()/Remove() elsewhere in the project to trigger signaling.

Can you elaborate on that last point? `AddWatch` and `RemoveWatch` are defined as idempotent and `RemoveWatch` fails gracefully.

I'm ok making a more conservative change but I want to make sure I understand the danger.

Reply:
I don't think there is any [clear and present danger](https://en.wikipedia.org/wiki/Clear_and_Present_Danger_(film)). I just prefer defaulting to a uniform design pattern where possible to avoid confusion down the line if the two philosophies start to conflict.

Reply:
:)

Updated logic and tests to capture the second case you raised, please take a look.